pet.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+from subprocess import *
+import os
+import signal
+import click
+
+
+cli = click.Group()
+
+
+@cli.command()
+@click.argument('name')
+def start(name):
+    """starts new project"""
+    if os.path.exists("./projects/%s" % name):
+        if not os.path.isfile("./projects/%s/_lock" % name):
+            open("./projects/%s/_lock" % name, 'a')
```

Comment:
Once you use context manager, managing lock file would be handled in one layer. Which is a lot nicer approach.